torchvision/transforms/functional.py

```diff
@@ -299,17 +321,22 @@ def resize(img: Tensor, size: List[int], interpolation: int = Image.BILINEAR) ->
             :math:`\left(\text{size} \times \frac{\text{height}}{\text{width}}, \text{size}\right)`.
             In torchscript mode size as single int is not supported, use a tuple or
             list of length 1: ``[size, ]``.
-        interpolation (int, optional): Desired interpolation enum defined by `filters`_.
-            Default is ``PIL.Image.BILINEAR``. If input is Tensor, only ``PIL.Image.NEAREST``, ``PIL.Image.BILINEAR``
-            and ``PIL.Image.BICUBIC`` are supported.
+        interpolation (InterpolationModes): Desired interpolation enum defined by
+            :class:`torchvision.transforms.InterpolationModes`.
+            Default is ``InterpolationModes.BILINEAR``. If input is Tensor, only ``InterpolationModes.NEAREST``,
+            ``InterpolationModes.BILINEAR`` and ``InterpolationModes.BICUBIC`` are supported.
 
     Returns:
         PIL Image or Tensor: Resized image.
     """
+    if not isinstance(interpolation, InterpolationModes):
```

Comment:
This makes for a big BC-breaking change, as all other code that users had before won't work anymore because of this.

I think it might be worth guarding against `interpolation` being an `int`, something like
```python
if not isinstance(interpolation, InterpolationModes):
    # wrap in an Enum
```
otherwise IIUC all code from before won't work anymore, which I think is not something we can do.

Reply:
Yes, makes senses. Let me see if we can do something with that.